File: Rust/Medium/Accepted/number-of-dice-rolls-with-target-sum_3.rs
impl Solution {
    pub fn num_rolls_to_target(n: i32, k: i32, target: i32) -> i32 {
        
        let mut dp = vec![vec![0; target as usize + 1]; n as usize + 1];
        dp[0][0] = 1;
        for i in 1..=n as usize {
            for j in 1..=target as usize {
                for t in 1..=k as usize {
                    if j >= t {
                        dp[i][j] = (dp[i][j] + dp[i - 1][j - t]) % (1_000_000_000 + 7);
                    }
                }
            }
        }
        dp[n as usize][target as usize]
    }
}

File: C++/Medium/Accepted/number-of-dice-rolls-with-target-sum_3.cpp
class Solution {
public:
    int numRollsToTarget(int n, int k, int target) {
        
        int M = 1e9 + 7;
        vector<vector<int>> dp(n + 1, vector<int>(target + 1, 0));
        dp[0][0] = 1;
        for (int i = 1; i <= n; ++i) {
            for (int j = 1; j <= target; ++j) {
                for (int d = 1; d <= k; ++d) {
                    if (j < d) {
                        break;
                    }
                    dp[i][j] = (dp[i][j] + dp[i - 1][j - d]) % M;
                }
            }
        }
        return dp[n][target];
    }
};